test/integration/webhook.test.ts

```diff
@@ -41,4 +41,20 @@ describe('webhooks', () => {
       msg: 'Go to https://github.com/settings/apps/YOUR_APP and verify that the Webhook secret matches the value of the WEBHOOK_SECRET environment variable.'
     }))
   })
+
+  test("should not display signature mismatch error when another bot sends a payload", async () => {
+    await request(probot.server)
+      .post("/")
+      .send(data)
+      .set("x-github-event", "pull_request")
+      .set("x-hub-signature", "sha1=68f2d3f057a090d1e0d937585a882d3526c7ac3b")
+      .set("x-github-delivery", "d159ef00-247f-11ea-8d08-0862e7465cb1")
+      .expect(400);
+
+    expect(logger).not.toHaveBeenCalledWith(
+      expect.objectContaining({
+        msg: "signature does not match event payload and secret"
+      })
+    );
+  });
```

Comment:
the signature is calculated based on the event payload (`data`). So in this case it is to be expected that the signature is not matching the hash based on payload + secret. You have to use your own payload

Reply:
That makes sense, yeah. But why do some payloads coming from GitHub not match the signature then? Only payloads that come from GitHub, sent by GitHub bots (like `renovate`, for example)

Reply:
> But why do some payloads coming from GitHub not match the signature then

That's the million dollar question. I don't know. The error keeps popping up for people and I don't know where the problem lies.

Reply:
Is there a public GitHub API issue tracker that we could post this too, and follow along for when the fix goes out?

Reply:
https://github.community/ would probably be the best place to bring that up

Reply:
@gr2m could it be that the payload is getting mutated before the signature is checked? By some middleware or something (I am not too familiar with Probot's internals).

Reply:
I don't think so, because it happens very rarely. It rather think it's some weird edge case where the signature calculation in Node.js is not doing the same that the ruby implementation on GitHub's site